src/DynamoCoreWpf/Views/PackageManager/Controls/NumericUpDownControl.xaml.cs

```diff
@@ -0,0 +1,169 @@
+
+
+
+        // The watermark that will be displayed if no value is provided
+        public string Watermark
+        {
+            get { return (string)GetValue(WatermarkProperty); }
+            set { SetValue(WatermarkProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for Watermark.  
+        public static readonly DependencyProperty WatermarkProperty =
+            DependencyProperty.Register("Watermark", typeof(string), typeof(NumericUpDownControl),
+                new FrameworkPropertyMetadata("0"));
+
+        // The Value of the numerical up/down control. Bind to this property
+        public int Value
+        {
+            get { return (int)GetValue(ValueProperty); }
+            set { SetValue(ValueProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for Value. 
+        public static readonly DependencyProperty ValueProperty =
+            DependencyProperty.Register("Value",
+                typeof(int),
+                typeof(NumericUpDownControl),
+                new FrameworkPropertyMetadata(0, FrameworkPropertyMetadataOptions.BindsTwoWayByDefault, new PropertyChangedCallback(OnValuePropertyChanged)));
+
+        // Setting the input TextBox 
+        private static void OnValuePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var numericUpDownControl = d as NumericUpDownControl;
+            numericUpDownControl.inputField.Text = e.NewValue.ToString();
+            if (numericUpDownControl.watermarkLabel.Visibility == Visibility.Visible)
+            {
+                numericUpDownControl.watermarkLabel.Visibility = Visibility.Collapsed;
+            }
+        } 
+        #endregion
+
+        public NumericUpDownControl()
+        {
+            InitializeComponent();
+
+            _numMatch = new Regex(@"^\d+$");
+        }
+
+        #region UI utility functions
+        private void spinnerUp_Click(object sender, RoutedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(inputField.Text))
+            {
+                if (Int32.TryParse(watermarkLabel.Content as string, out int watermarkValue))
+                {
+                    Value = ++watermarkValue;
+                    return;
+                }
+                else
+                {
+                    inputField.Text = "0"; // increment from 0 if Watermark is anyhting else but integer
+                }
+            }
+            if (Int32.TryParse(inputField.Text, out int value))
+            {
+                Value = ++value;
+            };
+        }
+
+        private void spinnerDown_Click(object sender, RoutedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(inputField.Text))
+            {
+                if (Int32.TryParse(watermarkLabel.Content as string, out int watermarkValue))
+                {
+                    Value = --watermarkValue;
```

Comment:
could we ever go to negative here?

Reply:
Changed to only allows positive whole numbers.